packages/did/src/DidResolver/DidResolver.ts

```diff
@@ -64,16 +63,16 @@ export async function resolveDoc(
       return null
     }
     case 'light': {
-      let details: LightDidDetails
+      let details: DidDetails
       try {
-        details = LightDidDetails.fromUri(did, false)
+        details = Did.parseDetailsFromLightDid(did, false)
       } catch (cause) {
         throw new SDKErrors.InvalidDidFormatError(did, {
           cause: cause as Error,
         })
       }
 
-      const fullDidDetails = await queryDetails(details.identifier)
+      const fullDidDetails = await queryDetails(address)
```

Comment:
Looks like there’s a bug on one of the next lines here. Input is light DID, we parse details from it, find that it has been upgraded, but still return the original parsed details. Shouldn’t we return the details fetched from the chain, @tjwelde?

Reply:
I think I had a discussion with @ntn-x2 about this before. I think it was about, that we want to inform the user, that the did was upgraded. This information is obtained from the metadata.
Not sure whether to return the full did details together with that metadata information though. Sounds to be the best way to go, I find.

Reply:
Yes, there will be a value called `canonicalId` if a light DID was requested but full DID found. For the current implementation the application developer should resolve a light DID, check for presence of canonicalId, and then resolve the canonicalId again. I don’t think Sporran or SKYC do that, the assumption there was that the resolve call will simply produce the latest data even if an upgrade has happened.

The naive fix has a hurdle though. The DID resolution spec says whatever ID is queried, it MUST return a document with the same ID. I haven’t yet decided on a solution for that.

Reply:
Looks like this is bigger than a simple fix, let’s address it separately in https://github.com/KILTprotocol/ticket/issues/2163